arviz/data/io_pyro.py

```diff
@@ -45,10 +57,15 @@ def __init__(
             self.nchains = self.ndraws = 0
         self.prior = prior
         self.posterior_predictive = posterior_predictive
+        self.predictions = predictions
         self.coords = coords
         self.dims = dims
+        self.pred_dims = pred_dims
         import pyro
 
+        if self.predictions is not None and self.pred_dims is None:
+            raise ValueError("Prediction dims are needed for predictions group.")
```

Comment:
dims are not needed, if not present they will get some default names by ArviZ internal functions.

Reply:
But the thing is, the `predictions` variables have the same name as the `posterior_predictive` variables, so it uses the `posterior_predictive` coords which has different value and causes error.

Reply:
I am not sure I follow, predictions should not have access to `dims` at any point, only to `pred_dims`. Then the possible cases are:
* pred dims is None -> default dim and coord names
* pred dims is a dict -> use pred dims

Reply:
Yes, both predictions and predictions_constant_data use `pred_dims` instead of `dims`. I have raised that error just in case the user passes the predictions but does not pass `pred_dims`.

All other groups are using the default `dims`, only the `predictions` and `predictions_constant_data` groups need `pred_dims` because their variable names are same as `posterior_predictive` and `constant_data` respectively but their data has actually different dimensions. So they cannot use the default `dims` in any case. You can see an example [here](https://github.com/nitishp25/arviz-playground/blob/master/arviz-pyro-example.ipynb).

Are we on the same page now? Other groups will always use the default `dims` but predictions and predictions_constant_data must always use `pred_dims` right?

Reply:
Yes, my point is that dims can be None, which makes ArviZ use some defaults, pred_dims should have the option of being none and generating default values, not using dims, generating the defaults that correspond to the dataset. I am not sure what I am missing, I thought default dims were generated on a dataset basis, not on an inference data basis

Reply:
Ohh yes dims are generated on dataset basis.

> * pred dims is None -> default dim and coord names

Sorry I got lost here, thought that you were talking about `self.dims` :)